### PR TITLE
Issue Definition designer Automation test

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
1. IssuesAdminTest.testCommentSortDirection
2. IssueAdminTest.testRelatedIssuesComments
3. IssueAdminTest.testProtectedFields
4. IssuesTest.defaultAssignedToTest

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
The `newElementCache` methods should remain protected, not sure why they're getting switched to public here.

File: src/org/labkey/test/tests/issues/IssuesAdminTest.java [hunk 1]
```diff
@@ -154,19 +167,102 @@ public void customIssueNameTest()
...
+        goToProjectHome();
+        clickAndWait(Locator.linkWithText(ISSUE_LIST_NAME));
+        IssuesAdminPage issuesAdminPage = _issuesHelper.goToAdmin();
+
+        log("Checking for the protected field");
+        DomainFormPanel domainFormPanel = issuesAdminPage.getFieldsPanel();
+        checker().verifyTrue("Title should be disabled", domainFormPanel.getField("Title").isFieldProtected());
+        checker().verifyTrue("NotifyList should be disabled", domainFormPanel.getField("NotifyList").isFieldProtected());
+        checker().verifyTrue("AssignedTo should be disabled", domainFormPanel.getField("AssignedTo").isFieldProtected());
+        checker().verifyTrue("Resolution should be disabled", domainFormPanel.getField("Resolution").isFieldProtected());
```
Comment (cnathe) [inline src/org/labkey/test/tests/issues/IssuesAdminTest.java]: may be worth adding a couple of checks that some of the other default fields are not protected

File: src/org/labkey/test/tests/issues/IssuesTest.java [hunk 2]
```diff
@@ -820,14 +819,28 @@ public void defaultAssignedToTest()
...
+        //No NPE when the default user is selected.
         clickProject(getProjectName());
         waitAndClickAndWait(Locator.linkContainingText(ISSUE_SUMMARY_WEBPART_NAME));
         clickButton("New Issue");
-        // NPE
-        //clickButton("Cancel");
+        clickButton("Cancel");
 
-        // TODO: extend test to check validate full user selection list based on group selection...
         // TODO: compare user dropdown list between admin and new issues page
```
Comment (cnathe) [inline src/org/labkey/test/tests/issues/IssuesTest.java:828]: is this TODO still needed or is this covered in the check at line 843? maybe I am misreading it and there is more to do.

**Review by cnathe — COMMENTED** [review]
From a test case point-of-view, this looks to cover the items that we discussed. Thanks.

File: src/org/labkey/test/tests/issues/IssuesAdminTest.java (hunk 1 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/issues/IssuesAdminTest.java]: Done

File: src/org/labkey/test/tests/issues/IssuesTest.java (hunk 2 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/issues/IssuesTest.java:828]: I just read the second part or TODO worth adding  the test for "extend test to check validate full user selection list based on group selection" ?


> is this TODO still needed or is this covered in the check at line 843? maybe I am misreading it and there is more to do.

File: src/org/labkey/test/components/domain/DomainFieldRow.java [hunk 3]
```diff
@@ -526,7 +536,8 @@ public DomainFieldRow setConceptLabelField(String labelField)
     public ConceptPickerDialog clickExpectedVocabulary()
     {
         expand();
-        elementCache().expectedVocabularyButton().click();;
+        elementCache().expectedVocabularyButton().click();
+        ;
```
Comment (labkey-tchad) [inline src/org/labkey/test/components/domain/DomainFieldRow.java]:
```suggestion
        elementCache().expectedVocabularyButton().click();
```

**Review by labkey-tchad — APPROVED** [review]

File: src/org/labkey/test/tests/issues/IssuesTest.java (hunk 2 above)
Comment (cnathe) [reply src/org/labkey/test/tests/issues/IssuesTest.java:828]: Got it.

**Review by cnathe — APPROVED** [review]

**Pull request merged**